src/lib.rs

```diff
@@ -1468,6 +1465,84 @@ pub extern "C" fn zendoo_batch_verify_proofs_by_id(
     Box::into_raw(Box::new(ret))
 }
 
+/**
+ * This function is basically a clone of zendoo_batch_verify_proofs_by_id(),
+ * the only difference is the call to batch_verify_subset_with_segment_size()
+ * instead of batch_verify_subset().
+ * It is used for testing purpose only.
+ */
+#[no_mangle]
+pub extern "C" fn zendoo_batch_verify_proofs_by_id_with_segment_size(
+    batch_verifier: *const ZendooBatchVerifier,
+    ids_list: *const u32,
+    ids_list_len: usize,
+    prioritize: bool,
+    segment_size: usize,
```

Comment:
Can't we re-use some code among the zendoo_batch_verify_proofs_by_id_with_segment_size and zendoo_batch_verify_proofs_by_id like you did for the internal functions ?